plutus-core/plutus-core/src/PlutusCore/Default/Builtins.hs

```diff
@@ -172,10 +176,14 @@ instance uni ~ DefaultUni => ToBuiltinMeaning uni DefaultFun where
         makeBuiltinMeaning
             ((==) @Integer)
             (runCostingFunTwoArguments . paramEqualsInteger)
-    toBuiltinMeaning Concatenate =
+    toBuiltinMeaning AppendByteString =
         makeBuiltinMeaning
             BS.append
-            (runCostingFunTwoArguments . paramConcatenate)
+            (runCostingFunTwoArguments . paramAppendByteString)
+    toBuiltinMeaning ConsByteString =
+        makeBuiltinMeaning
+            (\n xs -> BS.cons (fromIntegral @Integer n) xs)
```

Comment:
The thing you're consing on will wrap round if it's outside the `Word8` range, but I guess that's fine.